net/packet/packet_test.go

```diff
@@ -29,6 +30,52 @@ const (
 	Fragment = ipproto.Fragment
 )
 
+func fullHeaderChecksumV4(b []byte) uint16 {
+	s := uint32(0)
+	for i := 0; i < len(b); i += 2 {
+		if i == 10 {
+			// Skip checksum field.
+			continue
+		}
+		s += uint32(binary.BigEndian.Uint16(b[i : i+2]))
+	}
+	for s>>16 > 0 {
+		s = s&0xFFFF + s>>16
+	}
+	return ^uint16(s)
+}
+
+func TestHeaderChecksums(t *testing.T) {
+	// This is not a good enough test, because it doesn't
+	// check the various packet types or the many edge cases
+	// of the checksum algorithm. But it's a start.
+
+	tests := []struct {
+		name   string
+		packet []byte
+	}{
+		{
```

Comment:
Here's a UDP response for a DNS query to `clients4.google.com`:

```
0000   02 00 45 00 00 74 e2 85 00 00 40 11 96 b5 64 64   ..E..t....@...dd
0010   64 64 64 42 d4 33 00 35 ec 55 00 60 d9 19 ed fd   dddB.3.5.U.`....
0020   81 80 00 01 00 02 00 00 00 01 08 63 6c 69 65 6e   ...........clien
0030   74 73 34 06 67 6f 6f 67 6c 65 03 63 6f 6d 00 00   ts4.google.com..
0040   01 00 01 c0 0c 00 05 00 01 00 00 01 1e 00 0c 07   ................
0050   63 6c 69 65 6e 74 73 01 6c c0 15 c0 31 00 01 00   clients.l...1...
0060   01 00 00 01 1e 00 04 8e fa bd ce 00 00 29 02 00   .............)..
0070   00 00 00 00 00 00                                 ......

```
As hex:
```
020045000074e2850000401196b5646464646442d4330035ec550060d919edfd8180000100020000000108636c69656e74733406676f6f676c6503636f6d0000010001c00c000500010000011e000c07636c69656e7473016cc015c031000100010000011e00048efabdce0000290200000000000000
```


The first two bytes are wiresharl framing which can be trimmed.

Reply:
Added